File: src/network_calc.py
from requests import get

class NetworkCalc:
	def __init__(self) -> None:
		self.api = "https://networkcalc.com/api"
		self.headers = {
			"user-agent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/97.0.4692.99 Safari/537.36",
		}

	def calculate_subnet(
			self,
			ip_address: str,
			binary: bool = True,
			fields: str = "network_address,broadcast_address") -> dict:
		return get(
			f"{self.api}/ip/{ip_address}/27?binary={binary}&fields=[{fields}]",
			headers=self.headers).json()

	def convert_binary(
			self,
			binary: str,
			convert_from: int,
			convert_to: int) -> dict:
		return get(
			f"{self.api}/binary/{binary}?from={convert_from}&to={convert_to}",
			headers=self.headers).json()

	def get_website_certificate(self, hostname: str) -> dict:
		return get(
			f"{self.api}/security/certificate/{hostname}",
			headers=self.headers).json()

	def get_dns_records(self, hostname: str) -> dict:
		return get(
			f"{self.api}/dns/lookup/{hostname}",
			headers=self.headers).json()

	def get_whois_registration(self, hostname: str) -> dict:
		return get(
			f"{self.api}/dns/whois/{hostname}",
			headers=self.headers).json()

	def get_spf_record(self, hostname: str) -> dict:
		return get(
			f"{self.api}/dns/spf/{hostname}",
			headers=self.headers).json()

	def encode_value(self, value: str) -> dict:
		return get(
			f"{self.api}/api/encoder/{value}",
			headers=self.headers).json()

	def decode_value(self, encoding: str) -> dict:
		return get(
			f"{self.api}/api/encoder/{value}?encoding={encoding}&decode=True",
			headers=self.headers).json()

File: src/network_calc.js
class NetworkCalc {
	constructor() {
		this.api = "https://networkcalc.com/api"
		this.headers = {
			"User-Agent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/97.0.4692.99 Safari/537.36"
		}
	}

	async calculateSubnet(ipAddress, binary = true, fields = "network_address,broadcast_address") {
		const response = await fetch(
			`${this.api}/ip/${ipAddress}/27?binary=${binary}&fields=[${fields}]`, {
				method: "GET",
				headers: this.headers
			})
		return response.json()
	}

	async convertBinary(binary, convertFrom, convertTo) {
		const response = await fetch(
			`${this.api}/binary/${binary}?from=${convertFrom}&to=${convertTo}`, {
				method: "GET",
				headers: this.headers
			})
		return response.json()
	}

	async getWebSiteCertificate(hostName) {
		const response = await fetch(
			`${this.api}/security/certificate/${hostName}`, {
				method: "GET",
				headers: this.headers
			})
		return response.json()
	}

	async getDnsRecords(hostName) {
		const response = await fetch(
			`${this.api}/dns/lookup/${hostName}`, {
				method: "GET",
				headers: this.headers
			})
		return response.json()
	}

	async getWhoIsRegistration(hostName) {
		const response = await fetch(
			`${this.api}/dns/whois/${hostName}`, {
				method: "GET",
				headers: this.headers
			})
		return response.json()
	}

	async getSpfRecord(hostName) {
		const response = await fetch(
			`${this.api}/dns/spf/${hostName}`, {
				method: "GET",
				headers: this.headers
			})
		return response.json()
	}

	async encodeValue(value) {
		const response = await fetch(
			`${this.api}/api/encoder/${value}`, {
				method: "GET",
				headers: this.headers
			})
		return response.json()
	}

	async decodeValue(value, encoding) {
		const response = await fetch(
			`${this.api}/api/encoder/${value}?encoding=${encoding}&decode=True`, {
				method: "GET",
				headers: this.headers
			})
		return response.json()
	}
}

module.exports = {NetworkCalc}
